hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/BaseSparkCommitActionExecutor.java

```diff
@@ -157,6 +157,9 @@ public HoodieWriteMetadata<HoodieData<WriteStatus>> execute(HoodieData<HoodieRec
 
     // Handle records update with clustering
     HoodieData<HoodieRecord<T>> inputRecordsWithClusteringUpdate = clusteringHandleUpdate(inputRecords);
+    if (config.getBuildWorkloadParallelism() > 0) {
+      inputRecordsWithClusteringUpdate = inputRecordsWithClusteringUpdate.repartition(config.getBuildWorkloadParallelism());
```

Comment:
@jonvex do you have intreast to review this minor change?

Reply:
> Refer to the initial file number which is to upsert to hudi table, we can change the number by change source file number but the way is not convenient

@xuzifu666 got it, and this is an existing question in other logical, add new param is not friendly to user, I have a pr will optimizing the inference problem of partitions and it can also solve your problem

Reply:
OK，if have a patch we can try it in our condition.

Reply:
> and this is an existing question in other logical, add new param is not friendly to user, I have a pr will optimizing the inference problem of partitions and it can also solve your problem

This would be nice, the fix looks promising.